### DISCUSSION TITLE
Announce indexing after restart indexing

### DISCUSSION THREAD

**Discussion by WayneWayner** [subquery/subql#962] [Frontier TestNet - Bug Reporting]
There are 3 states for indexing deployments:

- Not indexing
- Indexing 
- Ready

However, if you **restart indexing,** you are still in the **ready** state and cannot switch to the **indexing** state. Only **not indexing** would be possible if you have no active service agreements. 
Would it be possible to add the transition **ready -> indexing**?

**Comment by mzxyz** [comment]
Good question.

1. If indexer has ongoing service agreement, indexer should make sure to provide stable indexing service. So indexer can not change the status back from READY to indexing. If indexer need to restart indexing from the beginning,  but better way is running the same service in another place before stop the current one which can make the switch without gap. 

The last upgrade in season2 which includes breaking change need the indexer to reindexing the project from beginning with inconsistent network status, this is for convenient in the TestNet.
